packages/firebase_analytics/firebase_analytics/lib/src/firebase_analytics.dart

```diff
@@ -1171,7 +1366,14 @@ class FirebaseAnalytics extends FirebasePluginPlatform {
     String? transactionId,
     String? affiliation,
     List<AnalyticsEventItem>? items,
+    Map<String, Object?>? parameters,
   }) {
+    _assertParameterTypesAreCorrect(parameters);
+
+    items?.forEach((item) {
```

Comment:
Hey @muccy, you may also create a method like `_assertItemsParameterTypesAreCorrect(items)` to avoid duplicating the following 3 lines in all these methods.